daemon/settings.c

```diff
@@ -507,6 +525,29 @@ settings_userdata_path(uid_t uid)
                            (unsigned)uid);
 }
 
+static void
+settings_remove_stale_userdata(uid_t uid)
+{
+    gchar *path = settings_userdata_path(uid);
+    if( g_file_test(path, G_FILE_TEST_EXISTS) ) {
+        if( unlink(path) == -1 )
+            log_err("%s: could not remove: %m", path);
+    }
+    g_free(path);
+}
+
+static bool
+settings_valid_user(const settings_t *self, uid_t uid)
+{
+    const control_t *control = settings_control(self);
+    /* Settings considers guest user invalid if it doesn't have an active user
+     * session. Settings for guest user are never stored on disk.
```

Comment:
This is fine as-is, but I'm trying to think if there is some reason stopping us from making this a general rule i.e. have the guest user considered already in control_valid_user() / users_user_exists()?

Reply:
I guess the only reason would that it might be confusing to get "invalid user" from D-Bus when the user actually exists. If that's okay or handled some other way, then it's fine.

Reply:
Now it says:
```
Invalid application name: sailfish-office
```
Which might not be any better...

Reply:
Moved this logic to control_valid_user. Now it replies:
Invalid user id: 105000
Which is probably better although still a little bit off.

Reply:
Ok, now it's "Guest user is not logged in" but that doesn't check if the user exists. I think that would be just unnecessary complication if it was added there.

Reply:
Looks just fine now